platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/events/performance/JankStatsMonitor.kt

```diff
@@ -0,0 +1,81 @@
+// capture-sdk - bitdrift's client SDK
+// Copyright Bitdrift, Inc. All rights reserved.
+//
+// Use of this source code is governed by a source available license that can be found in the
+// LICENSE file or at:
+// https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt
+
+package io.bitdrift.capture.events.performance
+
+import android.view.Window
+import androidx.metrics.performance.FrameData
+import androidx.metrics.performance.JankStats
+import io.bitdrift.capture.LogLevel
+import io.bitdrift.capture.LogType
+import io.bitdrift.capture.LoggerImpl
+import io.bitdrift.capture.common.Runtime
+import io.bitdrift.capture.common.RuntimeFeature
+import io.bitdrift.capture.events.lifecycle.IWindowListener
+import io.bitdrift.capture.providers.toFields
+
+/**
+ * Reports Jank Frames and its duration in ms
+ *
+ * This will be a no-op when `client_feature.android.jank_stats_reporting` kill switch is disabled
+ */
+internal class JankStatsMonitor(
+    private val logger: LoggerImpl,
+    runtime: Runtime,
+) : IWindowListener,
+    JankStats.OnFrameListener {
+    private val shouldDisableMonitorViaKillSwitch = !runtime.isEnabled(RuntimeFeature.JANK_STATS_EVENTS)
+
+    private var jankStats: JankStats? = null
+    private var currentWindow: Window? = null
+
+    override fun onFrame(volatileFrameData: FrameData) {
+        if (volatileFrameData.shouldReportJankyFrame()) {
+            sendJankStatData(volatileFrameData)
+        }
+    }
+
+    /**
+     * Called when we have a new window available
+     */
+    override fun onWindowAvailable(window: Window) {
+        if (shouldDisableMonitorViaKillSwitch || currentWindow == window) {
+            return
+        }
+        currentWindow = window
+        jankStats = JankStats.createAndTrack(window, this)
```

Comment:
might need to handle error case. from their docs:
>Note: The window must be active with a non-null DecorView, otherwise it returns an IllegalStateException.